### PR TITLE
feat(prover): fix / upgrade integration test

### PR DESCRIPTION
**Pull request by mask-pp**
Split integration-test into 3 steps:
1. Shell scripts to control docker compose.
3. Scripts for deploy l1 contracts.
4. Integration-test scripts.
5. Replace `common.Hex2Bytes` with `common.FromHex`, so that we don't need to consider the "0x" prefix.

### CODE REVIEW

**Comment by davidtaikocha** [comment]
image

Looks like the github action is failed (but don't know why it says successful in this PR page...)

**Comment by mask-pp** [comment]
> image
> Looks like the github action is failed (but don't know why it says successful in this PR page...)
> 
> ref: https://github.com/taikoxyz/taiko-client/actions/runs/7486090636/job/20375748869

Because `check_command solc` is unnecessary, and has been removed in this [commit](https://github.com/taikoxyz/taiko-client/pull/496/commits/e24b77c4eb74b46f4c7b1983838520f3392300df).

File: docker/start.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+compose_up "${DOCKER_SERVICE_LIST[@]}"
+
+# check until l1 chain is ready
+until cast chain-id --rpc-url "$L2_EXECUTION_ENGINE_HTTP_ENDPOINT"; do
+    sleep 1
+done
+
+# show all the running containers
+echo
```
Comment (davidtaikocha) [inline docker/start.sh:17]: is this empty `echo` necessary?

File: docker/start.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+source scripts/common.sh
+source docker/docker_env.sh
+
+# start docker compose service list
+echo "start docker compose service: ${DOCKER_SERVICE_LIST[*]}"
+
+compose_up "${DOCKER_SERVICE_LIST[@]}"
+
+# check until l1 chain is ready
```
Comment (davidtaikocha) [inline docker/start.sh:11]:
```suggestion
# check until L1 chain is ready
```

File: integration_test/deploy_l1_contract.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+check_env "TAIKO_MONO_DIR"
+
+cd "$TAIKO_MONO_DIR"/packages/protocol &&
+ forge script script/DeployOnL1.s.sol:DeployOnL1 \
+  --fork-url "$L1_NODE_HTTP_ENDPOINT" \
+  --broadcast \
+  --ffi \
+  -vvvvv \
+  --private-key "$PRIVATE_KEY" \
+  --block-gas-limit 100000000
```
Comment (davidtaikocha) [inline integration_test/deploy_l1_contract.sh:18]:
```suggestion
  --block-gas-limit 100000000
  
```

File: integration_test/l1_env.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    "0x90F79bf6EB2c4f870365E785982E1f101E93b906"
+    "0x15d34AAf54267DB7D7c367839AAf71A00a2C6A65"
+)
+GUARDIAN_PROVERS_ADDRESSES=$(printf ",%s" "${GUARDIAN_PROVERS_ADDRESSES_LIST[@]}")
+export GUARDIAN_PROVERS=${GUARDIAN_PROVERS_ADDRESSES:1}
+export MIN_GUARDIANS=${#GUARDIAN_PROVERS_ADDRESSES_LIST[@]}
+
+# Get the hash of L2 genesis.
+export L2_GENESIS_HASH=$(cast block --rpc-url "$L2_EXECUTION_ENGINE_HTTP_ENDPOINT" 0x0 -f hash)
+echo "l2chain genesis hash: $L2_GENESIS_HASH"
```
Comment (davidtaikocha) [inline integration_test/l1_env.sh:30]:
```suggestion
echo "l2chain genesis hash: $L2_GENESIS_HASH"

```

File: scripts/common.sh [hunk 5]
```diff
@@ -25,3 +27,18 @@ check_command() {
...
+  echo "done"
+}
+
+compose_up() {
+  local services=("$@")
+  echo
+  echo "launching services..."
+  $COMPOSE up --quiet-pull "${services[@]}" -d --wait
+  echo "done"
+}
```
Comment (davidtaikocha) [inline scripts/common.sh:44]:
```suggestion
}

```

File: docker/start.sh (hunk 1 above)
Comment (mask-pp) [reply docker/start.sh:17]: This is optional in order to display an empty line.

File: integration_test/l1_env.sh (hunk 4 above)
Comment (davidtaikocha) [inline integration_test/l1_env.sh:30]:
```suggestion
echo "L2 chain genesis hash: $L2_GENESIS_HASH"
```

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**